x-pack/plugins/infra/public/components/infrastructure_node_metrics_tables/shared/types.ts

```diff
@@ -21,3 +21,24 @@ export interface SourceProviderProps {
 export type IntegratedNodeMetricsTableProps = UseNodeMetricsTableOptions &
   SourceProviderProps &
   CoreProvidersProps;
+
+export type NodeMetricsTableData<NodeMetricsRow> =
+  | {
+      state: 'unknown';
+    }
+  | {
+      state: 'no-indices';
+    }
+  | {
+      state: 'empty-indices';
+    }
+  | {
+      state: 'data';
```

Comment:
nit: Perhaps `state: success` will describe this object better. What to you think?

Reply:
But is "there are not data in the indices" not a success? The query successfully executed and returned a valid result.

Reply:
Fair point. I saw this more as successfully returned valid data.